File: frontend/src/lib/user-display.ts
/** Prefer full name for people-facing labels; username is login identity only. */
export function userDisplayName(user: {
  fullName?: string | null;
  username?: string | null;
} | null | undefined): string {
  if (!user) {
    return "—";
  }
  const fullName = user.fullName?.trim();
  if (fullName) {
    return fullName;
  }
  const username = user.username?.trim();
  if (username) {
    return username;
  }
  return "—";
}

File: backend/handlers/user_display.go
package handlers

import (
	"strings"

	"am-keramika-backend/dto"
	"am-keramika-backend/models"
)

// userDisplayName prefers full name for people-facing labels; username is login-only fallback.
func userDisplayName(fullName, username string) string {
	if name := strings.TrimSpace(fullName); name != "" {
		return name
	}
	return strings.TrimSpace(username)
}

func mapUserSummary(user models.User) *dto.UserSummaryResponse {
	if user.ID == 0 {
		return nil
	}
	return &dto.UserSummaryResponse{
		ID:       user.ID,
		Username: user.Username,
		FullName: strings.TrimSpace(user.FullName),
	}
}
